graphql/util.go

```diff
@@ -200,12 +200,18 @@ func getPatchProjectVariantsAndTasksForUI(ctx context.Context, apiPatch *restMod
 			DisplayName: buildVariant.DisplayName,
 		}
 		projTasks := []string{}
-		for _, taskUnit := range buildVariant.Tasks {
-			projTasks = append(projTasks, taskUnit.Name)
-		}
+		executionTasks := []string{}
 		for _, displayTask := range buildVariant.DisplayTasks {
 			projTasks = append(projTasks, displayTask.Name)
+			executionTasks = append(executionTasks, displayTask.ExecTasks...)
+		}
+		for _, taskUnit := range buildVariant.Tasks {
+			// Only add task if it is not an execution task.
+			if !utility.StringSliceContains(executionTasks, taskUnit.Name) {
```

Comment:
This is a bit expensive since we'll need to loop through all execution tasks worst case. Using a map[string]bool for executionTasks would be more performant.